Modules/SaltStackConfig/SaltStackConfig.psm1

```diff
@@ -0,0 +1,304 @@
+<#
+Copyright 2021 VMware, Inc.
+SPDX-License-Identifier: BSD-2-Clause
+#>
+Function Connect-SscServer {
+<#
+  .NOTES
+  ===========================================================================
+   Created by:	Brian Wuchner
+   Date:		November 27, 2021
+   Blog:		www.enterpriseadmins.org
+   Twitter:		@bwuch
+  ===========================================================================
+  .SYNOPSIS
+    This function will allow you to connect to a vRealize Automation SaltStack Config API.
+    A global variable will be set with the Servername & Cookie/Header value for use by other functions.
+  .DESCRIPTION
+    Use this function to create the cookie/header to connect to SaltStack Config
+  .EXAMPLE
+    PS C:\> Connect-SscServer -Server 'salt.example.com' -Username 'root' -Password 'VMware1!'
+    This will default to internal user authentication.
+  .EXAMPLE
+    PS C:\> Connect-SscServer -Server 'salt.example.com' -Username 'bwuchner' -Password 'MyPassword1!' -AuthSource 'LAB Directory'
+    This will use the 'Lab Directory' LDAP authentication source.
+#>
+  param(
+    [Parameter(Mandatory=$true)][string]$server,
+    [Parameter(Mandatory=$true)][string]$username,
+    [Parameter(Mandatory=$true)][ValidateNotNullOrEmpty()][string]$password,
+    [string]$AuthSource='internal'
+  )
+  
+  $loginBody = @{'username'=$username; 'password'=$password; 'config_name'=$AuthSource}
+  try {
+    $webRequest = Invoke-WebRequest -Uri "https://$server/account/login" -SessionVariable ws -Headers $header
+    $ws.headers.Add('X-Xsrftoken', $webRequest.headers.'x-xsrftoken')
+    $webRequest = Invoke-WebRequest -Uri "https://$server/account/login" -WebSession $ws -method POST -body (ConvertTo-Json $loginBody)
+    $webRequestJson = ConvertFrom-JSON $webRequest.Content
+    $global:DefaultSscConnection = New-Object psobject -property @{ "SscWebSession"=$ws; "SscServer"=$server; "ConnectionDetail"=$webRequestJson }
+    
+	# Return a few grains, like the Salt server & version; this will prove the connection worked & provide some context
+	(Get-SscMaster).ret.salt.grains | Select-Object Host, NodeName, SaltVersion, @{N='Authenticated';E={$global:DefaultSscConnection.ConnectionDetail.authenticated}}, 

```

Comment:
Why don't you add these as properties of the server object (instead of ConnectionDetail) and return the server object instead? Normally all PowerCLI connect cmdlets return the server object.

Reply:
I did this because I wanted to control how the output looked and thought this was the way.  I think a Format.ps1xml file is what I really wanted, I'll implement this for the next commit.